src/transcript/sha3.rs

```diff
@@ -0,0 +1,100 @@
+use std::marker::PhantomData;
+use sha3::{Shake256, digest::*};
```

Comment:
Why using `Shake256` instead of [Sha3_256](https://docs.rs/sha3/latest/sha3/struct.Sha3_256Core.html)???

Reply:
I presume is either for the capacity difference or the security difference after Berstein's discussions with NIST. 
Can you bring some light? I just don't know why we don't use sha256 instead.